-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f8497e/12768/summary.html
**COMMIT**: 5465d63a74c61af590b7ebb82f115f292efb5ff4
**CMSSW**: CMSSW_11_3_X_2021-02-08-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32847/12768/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/ME0PreRecoGaussianModel.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/ME0PreRecoNoSmearModel.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/ME0SimpleModel.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/SealModule.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/ME0SimpleModel.cc: In member function 'int ME0SimpleModel::getSimHitBx(const PSimHit*, CLHEP::HepRandomEngine*)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-02-08-1100/src/SimMuon/GEMDigitizer/src/ME0SimpleModel.cc:135:5: error: 'LogDebug' was not declared in this scope; did you mean 'gDebug'?
  135 |     LogDebug("ME0SimpleModel") << "checktime "
      |     ^~~~~~~~
      |     gDebug
gmake: *** [tmp/slc7_amd64_gcc900/src/SimMuon/GEMDigitizer/src/SimMuonGEMDigitizer/ME0SimpleModel.cc.o] Error 1
>> Building  shared library tmp/slc7_amd64_gcc900/src/SimMuon/GEMDigitizer/src/SimMuonGEMDigitizer/libSimMuonGEMDigitizer.so
</pre>

--- a/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
+++ b/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
@@ -10,6 +10,7 @@
 #include "FWCore/Framework/interface/global/EDProducer.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 #include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "DataFormats/Candidate/interface/CandidateFwd.h"
 #include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
 #include "SimDataFormats/CrossingFrame/interface/CrossingFrame.h"

--- a/SimCalorimetry/EcalSimAlgos/interface/EBHitResponse.icc
+++ b/SimCalorimetry/EcalSimAlgos/interface/EBHitResponse.icc
@@ -10,6 +10,7 @@
 #include "CLHEP/Random/RandGaussQ.h"
 #include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/Utilities/interface/isFinite.h"
 #include "FWCore/Utilities/interface/Exception.h"
 

--- a/SimCalorimetry/EcalTestBeam/src/EcalTBDigiProducer.cc
+++ b/SimCalorimetry/EcalTestBeam/src/EcalTBDigiProducer.cc
@@ -5,6 +5,7 @@
 #include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "Geometry/CaloGeometry/interface/CaloGeometry.h"
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "SimCalorimetry/EcalSimAlgos/interface/EBHitResponse.h"

--- a/SimG4CMS/Calo/src/HcalTestHistoManager.cc
+++ b/SimG4CMS/Calo/src/HcalTestHistoManager.cc
@@ -6,6 +6,7 @@
 #include "SimG4CMS/Calo/interface/HcalTestHistoManager.h"
 
 #include "FWCore/PluginManager/interface/PluginManager.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include <cmath>
 #include <iostream>

--- a/SimGeneral/PileupInformation/plugins/GenPUProtonProducer.cc
+++ b/SimGeneral/PileupInformation/plugins/GenPUProtonProducer.cc
@@ -17,6 +17,7 @@
 #include "SimGeneral/HepPDTRecord/interface/ParticleDataTable.h"
 #include "DataFormats/HepMCCandidate/interface/GenParticle.h"
 #include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include <vector>
 #include <string>

--- a/SimMuon/GEMDigitizer/src/ME0SimpleModel.cc
+++ b/SimMuon/GEMDigitizer/src/ME0SimpleModel.cc
@@ -5,6 +5,7 @@
 #include "Geometry/GEMGeometry/interface/ME0EtaPartitionSpecs.h"
 #include "Geometry/CommonTopologies/interface/TrapezoidalStripTopology.h"
 #include "Geometry/GEMGeometry/interface/ME0Geometry.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/Utilities/interface/Exception.h"
 #include "CLHEP/Random/RandFlat.h"
 #include "CLHEP/Random/RandPoissonQ.h"

--- a/SimMuon/MCTruth/src/GEMHitAssociator.cc
+++ b/SimMuon/MCTruth/src/GEMHitAssociator.cc
@@ -1,4 +1,5 @@
 #include "SimMuon/MCTruth/interface/GEMHitAssociator.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 using namespace std;
 

--- a/SimMuon/MCTruth/src/PSimHitMap.cc
+++ b/SimMuon/MCTruth/src/PSimHitMap.cc
@@ -1,5 +1,6 @@
 #include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
 #include "SimMuon/MCTruth/interface/PSimHitMap.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 void PSimHitMap::fill(const edm::Event &e) {
   theMap.clear();

--- a/SimMuon/MCTruth/src/RPCHitAssociator.cc
+++ b/SimMuon/MCTruth/src/RPCHitAssociator.cc
@@ -1,4 +1,5 @@
 #include "SimMuon/MCTruth/interface/RPCHitAssociator.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 using namespace std;
 

